### PR TITLE
Add the ability to parse signals based on a platform string

### PR DESCRIPTION
**Pull request by katiewasnothere**
This PR fixes a bug where parsing signals for LCOW would use windows signals, since the platform running is windows. Since unix packages do no build and cannot be used on windows, we cannot use the default signals and signal parsing in golang packages. Instead, we manually specify a list of unix signals to support, on par with the linux signals supported in docker [here](https://github.com/docker/docker-ce/blob/master/components/engine/pkg/signal/signal_linux.go). 

Added test to verify added support in this PR. 

Signed-off-by: Kathryn Baldauf <kabaldau@microsoft.com>

### CODE REVIEW

File: signals_windows.go [hunk 1]
```diff
@@ -25,37 +25,123 @@ import (
...
+}
+
+func ParsePlatformSignal(rawSignal, platform string) (syscall.Signal, error) {
+	if platform != "windows" {
+		return parseSignalFromMap(rawSignal, signalMapLinux)
+	}
+	return parseSignalFromMap(rawSignal, signalMapWindows)
+}
+
+func parseSignalFromMap(rawSignal string, signalList map[string]syscall.Signal) (syscall.Signal, error) {
```
Comment (dcantah) [inline signals_windows.go]: Any reason for the name change?

File: signals_windows.go [hunk 2]
```diff
@@ -25,37 +25,123 @@ import (
...
+	"RTMAX-3":  linuxSigrtmax - 3,
+	"RTMAX-2":  linuxSigrtmax - 2,
+	"RTMAX-1":  linuxSigrtmax - 1,
+	"RTMAX":    linuxSigrtmax,
 }
 
 // ParseSignal parses a given string into a syscall.Signal
 // the rawSignal can be a string with "SIG" prefix,
 // or a signal number in string format.
 func ParseSignal(rawSignal string) (syscall.Signal, error) {
```
Comment (dcantah) [inline signals_windows.go:122]: Is there going to be another PR to cri to change which function is called here https://github.com/jterry75/cri/blob/9863254e760f7363b2d0192b184b221abf5b84f6/pkg/server/container_stop.go#L128? Even if we were running an LCOW container, this function would get called as the host is Windows and then we've hardcoded `signalMapWindows` to be used here. If we received the signal that alerted us to the issue (SIGWINCH) we'd still error out here just the same so the above would have to be updated to call `ParsePlatformSignal`

File: signals_windows.go (hunk 1 above)
Comment (dcantah) [reply signals_windows.go]: `signalMap` to `signalList` as that wasn't very clear

File: signals_windows.go (hunk 2 above)
Comment (katiewasnothere) [reply signals_windows.go:122]: Yes there will be an additional PR after this one.

File: signals_windows.go (hunk 1 above)
Comment (katiewasnothere) [reply signals_windows.go]: Done, though changed to use `map` consistently, not `list`

Comment (dcantah) [reply signals_windows.go]: Sweet, thanks!

File: signals_windows.go (hunk 2 above)
Comment (dcantah) [reply signals_windows.go:122]: Gotcha 👍

File: signals.go [hunk 3]
```diff
@@ -48,10 +48,6 @@ func GetStopSignal(ctx context.Context, container Container, defaultSignal sysca
 
 // GetOCIStopSignal retrieves the stop signal specified in the OCI image config
 func GetOCIStopSignal(ctx context.Context, image Image, defaultSignal string) (string, error) {
-	_, err := ParseSignal(defaultSignal)
```
Comment (dcantah) [inline signals.go:51]: I'm paging back in this work but why was this removed? If it actually serves no purpose/can be removed with this work, can we remove the `defaultSignal` parameter since it's no longer used anywhere else in this call?

Comment (katiewasnothere) [reply signals.go:51]: I decided to update the function a bit to actually parse the signal for either the running platform or the platform of the image, so as to not break expected behavior with this (though it's only used in ctr so not a massive problem).

**Review by dcantah — APPROVED** [review]
LGTM
